manifests/plugin/tasks.pp

```diff
@@ -4,14 +4,41 @@
 #
 # === Parameters:
 #
-# $package:: Package name to install, use ruby193-rubygem-foreman-tasks on Foreman 1.8/1.9 on EL
+# $package::        Package name to install, use ruby193-rubygem-foreman-tasks on Foreman 1.8/1.9 on EL
+#
+# $service::        Service name
+#
+# $manage_config::  If we should manage the config file for foreman-tasks. Defaults to false
+#
+# $logger_dynflow:: Logging configuration for dynflow. Defaults to true
+#
+# $logger_action::  Logging configuration for action. Defaults to true
+#
+# $cleanup_after::  Cleaning configuration: how long should the actions be kept before deleted
+#                   by `rake foreman_tasks:clean` task. Defaults to undef
 #
 # $service:: Service name
 #
 class foreman::plugin::tasks(
-  $package = $foreman::plugin::tasks::params::package,
-  $service = $foreman::plugin::tasks::params::service,
+  $package        = $foreman::plugin::tasks::params::package,
+  $service        = $foreman::plugin::tasks::params::service,
+  $manage_config  = $foreman::plugin::tasks::params::manage_config,
+  $logger_dynflow = $foreman::plugin::tasks::params::logger_dynflow,
+  $logger_action  = $foreman::plugin::tasks::params::logger_action,
+  $cleanup_after  = $foreman::plugin::tasks::params::cleanup_after,
 ) inherits foreman::plugin::tasks::params {
+  if $manage_config {
```

Comment:
foreman-tasks.yaml takes precedence over foreman_tasks.yaml.

I did a little test and I set cleanup after to 1w in foreman-tasks.yaml and to 5d in foreman_tasks.yaml:

        [root@puppet ~]# foreman-rake foreman_tasks:cleanup:config
        The tasks will be deleted after 1w
        The following actions are configured to be deleted automatically after some time:
        name                                               delete after
        Actions::Foreman::Host::ImportFacts                30d
        Actions::Foreman::Puppetclass::Import              30d
        [root@puppet ~]# mv /etc/foreman/plugins/foreman-tasks.yaml /etc/foreman/plugins/foreman-tasks.yaml.old
        [root@puppet ~]# foreman-rake foreman_tasks:cleanup:config
        The tasks will be deleted after 5d
        The following actions are configured to be deleted automatically after some time:
        name                                               delete after
        Actions::Foreman::Host::ImportFacts                30d
        Actions::Foreman::Puppetclass::Import              30d

Also, should I try to setup a cron job for foreman_tasks:cleanup:run?